.github/workflows/docker.yaml

```diff
@@ -107,14 +87,19 @@ jobs:
       - name: Login to AWS ECR
         if: steps.cfg.outputs.needs-ecr
         uses: aws-actions/amazon-ecr-login@v2
-      - name: Publish Docker Images
+      - name: Build Docker Images
+        env:
+          BUILDX_GIT_LABELS: full
+          CACHE_TAG: ghcr.io/tenzir/tenzir:build-cache
         run: |
-          # Expand the registries arrays and output each entry together with
-          # the corresponding edition.
-          jq -r '.editions[] | .name as $name | .registries[]? | [ $name, . ] | join(" ")' \
-            <<< '${{ inputs.config }}' | while read -r name registry; do
-            for tag in ${{ join(fromJSON(inputs.config).tags, ' ') }}; do
-              docker tag "tenzir/${name}:latest" "${registry}/tenzir/${name}:${tag}"
-              docker push "${registry}/tenzir/${name}:${tag}"
-            done
+          cmake_version_args="-DTENZIR_VERSION_BUILD_METADATA:STRING=${{ fromJSON(inputs.config).version-build-metadata }}"
+          echo "::info Building the Developer Edition Image"
+          jq -r '.editions[] | .name as $name | .target as $target | .registries[]? | [ $name, $target, . ] | join(" ")' \
+            <<< '${{ inputs.config }}' | while read -r name target registry; do
+            docker buildx build . \
+              -t "${registry}/tenzir/${name}:latest" --target "${target}" \
+              --platform linux/amd64,linux/arm64 --push \
```

Comment:
Note to future self: This exceeds the 6 hour timeout that GitHub Actions workflows have. We need to instead build two separate images on their respective native platforms, and then in merge the two with `docker manifest create` manually.